src/tests/ftest/dfuse/pil4dfs_fio.py

```diff
@@ -0,0 +1,223 @@
+"""
+  (C) Copyright 2019-2024 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import json
+import os
+
+from ClusterShell.NodeSet import NodeSet
+from cpu_utils import CpuInfo
+from dfuse_test_base import DfuseTestBase
+from fio_utils import FioCommand
+from general_utils import bytes_to_human
+
+
+class Pil4dfsFio(DfuseTestBase):
+    """Test class Description: Runs Fio with in small config.
+
+    :avocado: recursive
+    """
+
+    _FIO_RW_NAMES = ["write", "read"]
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a FioPil4dfs object."""
+        super().__init__(*args, **kwargs)
+
+        self.fio_cmd = None
+        self.fio_params = {"thread": "", "blocksize": "", "size": ""}
+        self.fio_numjobs = 0
+        self.fio_cpus_allowed = ""
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+
+        # Start the servers and agents
+        super().setUp()
+
+        for name in self.fio_params:
+            self.fio_params[name] = self.params.get(name, "/run/fio/job/*", "")
+
+        cpu_info = CpuInfo(self.log, self.hostlist_clients)
+        cpu_info.scan()
+        _, arch = cpu_info.get_architectures()[0]
+        if arch.numas != 2:
+            self.fail(f"Client with unsupported quantity of NUMA nodes: wait=2, got={arch.numas}")
+        self.fio_numjobs = int(arch.quantity / arch.threads_core)
+        cpus = []
+        cores_quantity = int(self.fio_numjobs / 2)
+        for numa_idx in range(2):
+            cpus += arch.get_numa_cpus(numa_idx)[:cores_quantity]
+        self.fio_cpus_allowed = str(NodeSet(str(cpus)))[1:-1]
+
+    def create_container(self):
+        """Created a DAOS POSIX container"""
+        self.log.info("Creating pool")
+        self.assertIsNone(self.pool)
```

Comment:
Will you add a message for all these asserts? If this fails, it's not clear what the error is without digging into the logs
```suggestion
        self.assertIsNone(self.pool, "unexpected...")
```

Reply:
- [ ] Add relevant messages to assert

Reply:
Fixed with commit 4c600bcacb04f835b9ca971db94bf9dadb11c05f